packages/@aws-cdk/aws-ecs/test/ec2/ec2-task-definition.test.ts

```diff
@@ -1199,6 +1199,33 @@ describe('ec2 task definition', () => {
     });
   });
 
+  test('correctly sets inferenceAccelerators', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const inferenceAccelerator = [{
+      deviceName: 'device1',
+      deviceType: 'eia2.medium',
+    }];
+
+    const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'Ec2TaskDef', {
+      inferenceAccelerators: inferenceAccelerator,
+    });
+
+    taskDefinition.addContainer('web', {
+      image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),
+      memoryLimitMiB: 512,
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::ECS::TaskDefinition', {
+      Family: 'Ec2TaskDef',
+      InferenceAccelerators: [{
+        DeviceName: 'device1',
+        DeviceType: 'eia2.medium',
+      }],
+    });
+  });
+
```

Comment:
How about the case for erring out for setting inferenceAccelerators for fargate? (both through props and addInferenceAccelerator)